### ISSUE TITLE
Reproducibility issue for ROMS-Tools `Grid` when `RiverForcing` or `Nesting` is used

### ISSUE THREAD

**Issue by NoraLoose** [CWorthy-ocean/C-Star#201]
I want to surface a potential reproducibility issue in C-Star due to the way ROMS / ROMS-Tools handles river forcing and nesting.

A grid is created and saved via step 0:
```
grid = roms_tools.Grid(**kwargs)
grid.save(path)  # saves grid variables to NetCDF
grid.to_yaml(yaml_path)  # saves kwargs to YAML
```

In two instances (both of which are optional), `ROMS-Tools` modifies the grid file after the fact (because a modified grid file is what ROMS expects): 
1. When a `RiverForcing` class is created, a new variable `river_flux` gets added to the grid file, see [here](https://roms-tools.readthedocs.io/en/latest/river_forcing.html#River-locations). 
2. When a `Nesting` class is created via
```
nesting = roms_tools.Nesting(parent_grid=parent_grid, child_grid=child_grid)
nesting.save(path, child_grid_path)  # saves nesting information and updated child grid to NetCDF
``` 
the child grid **topography** and **mask** get modified  along the boundaries so that they are compatible with the parent grid, see [here](https://roms-tools.readthedocs.io/en/latest/nesting.html#The-Nesting-object).

In other words, there is a difference in the grid files depending on whether (optional) steps 1 and 2 are executed or not. Step 1 does not modify the existing grid variables (so it's not too bad), but step 2 modifies the existing grid variables.

Just reporting this here because this needs further discussion.
